# **Question 1:**

In [20]:
import pandas as pd
import numpy as np

In [21]:
orders_df = pd.read_excel('List_of_Orders.xlsx')
details_df = pd.read_excel('Order_Details.xlsx')
target_df = pd.read_excel('Sales_target.xlsx')

In [22]:
orders_df.head()

,Order ID,Order Date,CustomerName,State,City
0,B-25601,2018-01-04 00:00:00,Bharat,Gujarat,Ahmedabad
1,B-25602,2018-01-04 00:00:00,Pearl,Maharashtra,Pune
2,B-25603,2018-03-04 00:00:00,Jahan,Madhya Pradesh,Bhopal
3,B-25604,2018-03-04 00:00:00,Divsha,Rajasthan,Jaipur
4,B-25605,2018-05-04 00:00:00,Kasheen,West Bengal,Kolkata


In [23]:
merged_df = pd.merge(orders_df, details_df, on='Order ID')

In [24]:
category_metrics = merged_df.groupby('Category').agg(
    Total_Sales=('Amount', 'sum'),
    Total_Profit=('Profit', 'sum'),
    Unique_Order_Count=('Order ID', 'nunique')
).reset_index()

In [25]:
category_metrics['Avg_Profit_Per_Order'] = category_metrics['Total_Profit'] / category_metrics['Unique_Order_Count']
category_metrics['Profit_Margin_%'] = (category_metrics['Total_Profit'] / category_metrics['Total_Sales']) * 100

print("--- Part 1: Category Metrics ---")
print(category_metrics)

--- Part 1: Category Metrics ---
      Category  Total_Sales  Total_Profit  Unique_Order_Count  \
0     Clothing       139054         11163                 393   
1  Electronics       165267         10494                 204   
2    Furniture       127181          2298                 186   

   Avg_Profit_Per_Order  Profit_Margin_%  
0             28.404580         8.027817  
1             51.441176         6.349725  
2             12.354839         1.806874  


***Part 1: Sales and Profitability Analysis***

---



Top-performing Category: Clothing performed the best overall, with the highest profit margin (8.03%) and the most orders (393). Although Electronics generated the highest sales (₹165,267) and the highest profit per order (₹51.44), Clothing delivered more consistent and profitable results.

Underperforming Category: Furniture was the weakest performer. Despite sales of ₹127,181, it earned only ₹2,298 in profit, giving it a low profit margin of 1.81% and an average profit of just ₹12.35 per order.

Reason: Furniture likely has higher costs for shipping, storage, and discounts, which reduce profits. Clothing, on the other hand, usually has lower costs and better margins, making it more profitable overall.

In [26]:
furniture_target['Month of Order Date'] = pd.to_datetime(
    furniture_target['Month of Order Date']
)

furniture_target = furniture_target.sort_values(
    by='Month of Order Date'
).reset_index(drop=True)

furniture_target['MoM_Change_%'] = (
    furniture_target['Target'].pct_change() * 100
)

print(furniture_target[['Month of Order Date', 'Target', 'MoM_Change_%']])

   Month of Order Date  Target  MoM_Change_%
0           2026-01-19   11500           NaN
1           2026-02-19   11600      0.869565
2           2026-03-19   11800      1.724138
3           2026-04-18   10400    -11.864407
4           2026-05-18   10500      0.961538
5           2026-06-18   10600      0.952381
6           2026-07-18   10800      1.886792
7           2026-08-18   10900      0.925926
8           2026-09-18   11000      0.917431
9           2026-10-18   11100      0.909091
10          2026-11-18   11300      1.801802
11          2026-12-18   11400      0.884956


***Part 2: Target Achievement Analysis***

---



Month-over-Month Target Sales Analysis

The Furniture category targets increased gradually throughout the year. The highest drop occurred in April (-11.86%), where the target decreased from 11,800 to 10,400. After April, the targets increased steadily each month, with monthly growth between 0.87% and 1.89%, showing a stable trend.

Trend Insights and Recommendations

* The large drop in April indicates a significant change in target planning.
* Use historical sales data to set realistic monthly targets instead of applying uniform increases.
* Account for seasonality, such as festivals, holiday sales, or market trends, when assigning targets.
* Review targets quarterly and adjust them based on actual sales performance and market conditions.
* Investigate the April target drop to determine whether it was due to lower demand, inventory constraints, or a business decision.
* Forecast using predictive analytics (moving averages, time-series models) to create more accurate monthly targets.





In [27]:
state_orders = orders_df.groupby('State')['Order ID'].nunique().reset_index()
top_5_states = state_orders.sort_values('Order ID', ascending=False).head(5)['State'].tolist()

top_states_df = merged_df[merged_df['State'].isin(top_5_states)]

state_metrics = top_states_df.groupby('State').agg(
    Total_Sales=('Amount', 'sum'),
    Total_Profit=('Profit', 'sum'),
    Unique_Order_Count=('Order ID', 'nunique')
).reset_index()

state_metrics['Avg_Profit_Per_Order'] = state_metrics['Total_Profit'] / state_metrics['Unique_Order_Count']

state_metrics = state_metrics.set_index('State').loc[top_5_states].reset_index()

print("--- Part 3: Regional Performance (Top 5 States) ---")
print(state_metrics[['State', 'Total_Sales', 'Avg_Profit_Per_Order']])

--- Part 3: Regional Performance (Top 5 States) ---
            State  Total_Sales  Avg_Profit_Per_Order
0  Madhya Pradesh       105140             54.960396
1     Maharashtra        95348             68.622222
2       Rajasthan        21149             39.281250
3         Gujarat        21058             17.222222
4          Punjab        16786            -24.360000


***Part 3: Regional Performance Insights***

---



Regional Disparities:
Maharashtra delivers the highest average profit per order (68.62), while Punjab records a significant loss (-24.36), despite having a high number of orders.

Priority Areas for Improvement:

Punjab (Highest Priority): Investigate high logistics costs, excessive discounts, or product returns causing negative profitability.

Gujarat (Secondary Priority): Improve profit margins by adopting successful pricing and operational strategies used in Maharashtra.